packages/ubuntu_bootstrap/lib/slides/slides_provider.dart

```diff
@@ -0,0 +1,140 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:html/parser.dart' show parse;
+import 'package:intl/intl.dart' hide TextDirection;
+import 'package:intl/intl_standalone.dart';
+import 'package:path/path.dart' as path;
+import 'package:ubuntu_bootstrap/slides/slide_html.dart';
+import 'package:ubuntu_logger/ubuntu_logger.dart';
+import 'package:ubuntu_provision/ubuntu_provision.dart';
+
+final slidesProvider = Provider((ref) => SlidesModel());
+
+final _log = Logger('slides');
+
+/// Pre-caches and holds the HTML slides.
+class SlidesModel {
+  final List<SlideHtml> slides = [];
+
+  SlideHtml? get(int index) => slides[index];
+
+  Future<void> preCache() async {
+    final loadFutures = <Future<void>>[];
+    while (true) {
+      final index = loadFutures.length + 1;
+      final slidePath = '${ConfigService.whiteLabelDirectory}slides/$index';
+      final slideDirectory = Directory(slidePath);
+      if (!await slideDirectory.exists()) {
+        break;
+      }
+      loadFutures.add(_loadSlide(index));
+    }
+
+    if (loadFutures.isEmpty) {
+      _log.info('No custom slides found, using default slides.');
+      while (true) {
+        final index = loadFutures.length + 1;
+        final slideFuture = _loadSlide(index, fromAssets: true);
+        if (await slideFuture == null) {
+          break;
+        }
+        loadFutures.add(slideFuture);
+      }
+    }
+
+    slides.addAll((await Future.wait(loadFutures)).whereType());
+  }
+
+  Future<SlideHtml?> _loadSlide(int index, {bool fromAssets = false}) async {
+    const defaultLocale = 'en_US';
+    final locale = Intl.defaultLocale ?? await findSystemLocale();
```

Comment:
On the welcome page and in the storage dialogs we use `Localizations.localeOf(context)`. Is there a way we could do the same for the slides, or did you deliberately decide against that?

Reply:
We don't have the context in here, it would be the same result right?

Reply:
Ah, probably yes :+1: